comment/comment.go

```diff
@@ -59,6 +61,25 @@ func (m *GormCommentRepository) Create(ctx context.Context, u *Comment) error {
 	return nil
 }
 
+// Save a single comment
+func (m *GormCommentRepository) Save(ctx context.Context, comment *Comment) (*Comment, error) {
+	c := Comment{}
+	tx := m.db.Where("id=?", comment.ID).First(&c)
+	if tx.RecordNotFound() {
+		// treating this as a not found error: the fact that we're using number internal is implementation detail
+		return nil, errors.NewNotFoundError("comment", comment.ID.String())
+	}
+	if err := tx.Error; err != nil {
+		return nil, errors.NewInternalError(err.Error())
+	}
+	tx = tx.Save(comment)
```

Comment:
A comment should not be updated directly. We need to track the history of the conversation for audit.

The view should have a 'Comments' sections that show all the current version of the comments with indications where there has been change/deletion, and a 'Show complete history' which show all deleted and updated comments.